pkgs/development/libraries/rabbitmq-java-client/default.nix

```
@@ -1,52 +1,67 @@
 {
-  fetchurl,
   lib,
-  stdenv,
-  ant,
-  jdk,
-  jre,
-  python2,
-  makeWrapper,
+  maven,
+  fetchFromGitHub,
+  linkFarm,
+  python3,
+  ...
 }:
-
-stdenv.mkDerivation rec {
+let
+  rabbitmq-codegen = linkFarm "deps-dir" [
+    {
+      name = "rabbitmq_codegen";
+      path = fetchFromGitHub {
+        owner = "rabbitmq";
+        repo = "rabbitmq-codegen";
+        tag = "v3.8.10-beta.1";
+        hash = "sha256-rVA6zGnkzpFDUi5QkRHCA2wIg4Z96XW8XGqxYQycTms=";
+      };
+    }
+  ];
```

A `linkFarm` is unnecessary here; using the `fetchFromGitHub` derivation directly works just as well and is less complex.

**Additionally**, the rabbitmq-codegen repository has been archived, with the content being moved into the main rabbitmq-server repository:
> This was migrated to https://github.com/rabbitmq/rabbitmq-server

```suggestion
  rabbitmq-codegen = "${rabbitmq-server.src}/deps/rabbitmq_codegen";
```

---

`deps-dir` is expected to a folder in which `rabbitmq_codegen` lies, which makes it all the more odd that `rabbitmq-codegen = "${rabbitmq-server.src}/deps` causes the build process to fail.